src/backend/storage/file/buffile.c

```diff
@@ -1012,6 +1026,10 @@ BufFileStartCompression(BufFile *file)
 {
 	ResourceOwner oldowner;
 	size_t ret;
+	ZSTD_customMem customMem;
+
+	customMem.customAlloc = customAlloc;
+	customMem.customFree = customFree;
```

Comment:
I think that You also need to store CurrentResourceOwner and CurrentMemoryContext in the third field of the ZSTD_customMem structure, and then use them in callbacks from the first two fields.

Reply:
We switch memory contexts to assign then a current context to some tracked resource, don't we? For example, like in `zstd_alloc_context` where we set `ctx->owner = CurrentResourceOwner;`. But `ZSTD_CCtx` doesn't have such a field, so do we really need to keep track of these contexts?

Reply:
> `ZSTD_CCtx` doesn't have such a field

Yes, but ZSTD_customMem has a field and pass it in callbacks

Reply:
> do we really need to keep track of these contexts?

zstd_context uses TopMemoryContext, which means it expects a context switch.